sktime/clustering/partitioning/_lloyds.py

```diff
@@ -264,11 +265,18 @@ def _check_params(self, X: np.ndarray) -> None:
         self._random_state = check_random_state(self.random_state)
 
         if isinstance(self.init_algorithm, str):
-            self._init_algorithm = self._init_algorithms.get(self.init_algorithm)
-        else:
+            if self.init_algorithm not in self._init_algorithms.keys():
+                raise KeyError(
+                    f"The string provided for init_algorim: {self.init_algorithm} is "
+                    f"invalid. The following are a list of valid init algorithms "
+                    f"strings: {list(self._init_algorithms.keys())}"
+                )
+            self._init_algorithm = self._init_algorithms[self.init_algorithm]
+        elif isinstance(self.init_algorithm, np.ndarray):
```

Comment:
do you also want to check that it's 3D? and/or that it has the same number of variables (axis 1) as the data `X` seen here?

Reply:
I have add a check for `init_algorithm` begin 3D but I am not sure the best way to go about checking the same number of dimensions for X and `init_algorithm` without loads of `if` statements.

Reply:
that's not really blocking, fine with me.

I was simply thinking about `X.shape[1] == init_algorithm.shape[1]`, but perhaps that's "over-checking".

Reply:
addresed